src/Splat/ServiceLocation/Locator.cs

```diff
@@ -115,5 +120,27 @@ public static IDisposable RegisterResolverCallbackChanged(Action callback)
                 }
             });
         }
+
+        /// <summary>
+        /// This method will prevent resolver changed notifications from happening until
+        /// the returned <see cref="IDisposable"/> is disposed.
+        /// </summary>
+        /// <returns>A disposable which when disposed will indicate the change
+        /// notification is no longer needed.</returns>
+        public static IDisposable SuppressResolverCallbackChangedNotifications()
+        {
+            _resolverChangedNotificationSuspendCount++;
```

Comment:
Is there a reason you're not using Interlocked Increment/Decrement here?

Reply:
Using https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/volatile

I'm not needing atomic read/write operation in this case. You need interlocked if you need to do both the read wrote operation atomically. 

As per the c# standard. https://github.com/dotnet/csharplang/blob/master/spec/variables.md

```
Atomicity of variable references
Reads and writes of the following data types are atomic: bool, char, byte, sbyte, short, ushort, uint, int, float, and reference types. In addition, reads and writes of enum types with an underlying type in the previous list are also atomic. Reads and writes of other types, including long, ulong, double, and decimal, as well as user-defined types, are not guaranteed to be atomic. Aside from the library functions designed for that purpose, there is no guarantee of atomic read-modify-write, such as in the case of increment or decrement.
```

We are only ever needing a write then later a read. Never at the same time in this case.

Reply:
Yeah, but incrementing a variable requires three operations (read, increment, write) which is why Interlocked is recommended.

Reply:
Ah I'll update it's cool